pkg/R/RDD.R

```diff
@@ -1844,3 +1844,53 @@ setMethod("cogroup",
             cogroup.rdd <- mapValues(groupByKey(union.rdd, numPartitions), 
                                      group.func)
           })
+
+# TODO:
+# bind to env?
+#' Return an RDD's name.
+#'
+#' @param rdd The RDD whose name is returned.
+#' @rdname name
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, list(1,2,3))
+#' name(rdd) # NULL (if not set before)
+#'}
+setGeneric("name", function(rdd) { standardGeneric("name") })
+
+#' @rdname name
+#' @aliases name,RDD
+setMethod("name",
+          signature(rdd = "RDD"),
+          function(rdd) {
+            .jcall(getJRDD(rdd), "S", "name")
```

Comment:
I can see that "S" is a shortcut for "Ljava/lang/String;", but I'd actually prefer having the full type here to be explicit.